File: Ejercicios/4. WeatherData/Net/Before/WeatherData.Tests/DisplayTests.cs
using System;
using System.IO;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace WeatherData.Tests
{
    [TestClass]
    public class DisplayTests
    {
        private StringWriter sw;

        [TestInitialize]
        public void Setup()
        {
            this.sw = new StringWriter();
            Console.SetOut(this.sw);
        }

        [TestMethod]
        public void UpdateDisplay1()
        {
            var weatherData = new WeatherData();
            weatherData.SetMeasurements(80, 65, 30.4f);

            Assert.IsTrue(sw.ToString().Contains("Current conditions: 80F degrees and 65% humidity"));
            Assert.IsTrue(sw.ToString().Contains("Improving weather on the way!"));
            Assert.IsTrue(sw.ToString().Contains("Avg/Max/Min temperature = 80.00F/80F/80F")); 
        }

        [TestMethod]
        public void UpdateDisplay2()
        {
            var weatherData = new WeatherData();
            weatherData.SetMeasurements(82, 70, 29.2f);

            Assert.IsTrue(sw.ToString().Contains("Current conditions: 82F degrees and 70% humidity"));
            Assert.IsTrue(sw.ToString().Contains("Watch out for cooler, rainy weather"));
            Assert.IsTrue(sw.ToString().Contains("Avg/Max/Min temperature = 82.00F/82F/82F"));
        }

        [TestCleanup]
        public void TearDown()
        {
            this.sw.Dispose();
        }
    }
}

File: Ejercicios/4. WeatherData/Java/Before/src/test/weather/DisplayTests.java
package weather;

import static org.junit.Assert.*;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

public class DisplayTests {
	private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();

	@Before
	public void setUp() {
		System.setOut(new PrintStream(outContent));
	}

	@Test
	public void UpdateDisplay1() {
		WeatherData weatherData = new WeatherData();
		weatherData.setMeasurements(80, 65, 30.4f);

//		assertTrue(this.outContent.toString().contains(
//				"Current conditions: 80F degrees and 65% humidity"));
		assertTrue(this.outContent.toString().contains("Improving weather on the way!"));
		assertTrue(this.outContent.toString().contains("Avg/Max/Min temperature = 80.0F/80.0F/80.0F"));
	}

	@Test
	public void UpdateDisplay2() {
		WeatherData weatherData = new WeatherData();
		weatherData.setMeasurements(82, 70, 29.2f);

//		assertTrue(this.outContent.toString().contains(
//				"Current conditions: 82F degrees and 70% humidity"));
		assertTrue(this.outContent.toString().contains(
				"Watch out for cooler, rainy weather"));
		assertTrue(this.outContent.toString().contains(
				"Avg/Max/Min temperature = 82.0F/82.0F/82.0F"));
	}

	@After
	public void tearDown() {
		System.setOut(null);
	}

}
